EpikCord/__init__.py

```diff
@@ -1907,14 +1913,30 @@ async def connect(self, guild_id:str, channel_id:str, **kwargs):
                 "self_deaf" : kwargs.get("self_deaf", False)
             }
         })
-        #self.session_id = await self.client.ws.receive().json()
-        print(await (await self.client.ws.receive().json()))
-        print(await (await self.client.ws.receive()).json())
-        voice_update = await self.client.ws.receive().json()["d"]
-        self.token = voice_update["token"]
-        self.endpoint = voice_update["endpoint"]
-
+        def figure_out_receive(data):
+            if data["t"] == "VOICE_SERVER_UPDATE":
+                return "VOICE_SERVER_UPDATE"
+            elif data["deaf"]:
+                return "VOICE_STATE_UPDATE"
+            else:
+                return ""
+        first_receive, second_receive = await asyncio.gather(await self.client.ws.receive(),await self.client.ws.receive())
+        receive1 = first_receive.json()
+        receive2 = second_receive.json()
+        if figure_out_receive(receive1) == "VOICE_SERVER_UPDATE":
+            voice_server_update=receive1
+        elif figure_out_receive(receive2) == "VOICE_SERVER_UPDATE":
+            voice_server_update=receive2
         
+        if figure_out_receive(receive1) == "VOICE_STATE_UPDATE":
+            self.session_id=receive1["session_id"]
+        elif figure_out_receive(receive2) == "VOICE_STATE_UPDATE":
+            self.session_id=receive2["session_id"]
```

Comment:
I think you want to merge those 2 if and elif's

Reply:
And also what about that other event ?

Reply:
im revamping the system.